src/frequenz/client/reporting/_client.py

```diff
@@ -426,9 +428,7 @@ async def stream_method() -> (
             retry_strategy=None,
         )
 
-        receiver = broadcaster.new_receiver()
-        async for data in receiver:
-            yield data
+        return broadcaster.new_receiver()
```

Comment:
This appears to be creating a new `GrpcStreamBroadcaster` everytime a new receiver is requested.  But that would create a new connection to the API service each time.  Instead you should hold on to the `broadcaster` instance and reuse it, like here:
- https://github.com/frequenz-floss/frequenz-client-electricity-trading-python/blob/a029d214ea7cdaf7f84ed72c7990546705e7a558/src/frequenz/client/electricity_trading/_client.py#L210-L215
- https://github.com/frequenz-floss/frequenz-client-electricity-trading-python/blob/a029d214ea7cdaf7f84ed72c7990546705e7a558/src/frequenz/client/electricity_trading/_client.py#L299-L322